### PR TITLE
[chore] Miscellaneous Nits Before Release

### PR DESCRIPTION
**Pull request by patrick-ogrady**
This PR fixes a few miscellaneous issues found while integrating `rosetta-sdk-go` into `rosetta-cli` (https://github.com/coinbase/rosetta-cli/pull/193).

### Changes
- [x] Add `CounterStorage.GetTransactional`
- [x] Fix negative `big.Int` regression: https://play.golang.org/p/wD68BU8AACd
  - [x] find issue in docs: https://golang.org/pkg/math/big/#Int.Bytes
  - [x] reproduce with test
  - [x] don't store pruned if < 0 (otherwise will appear as positive when reading)
- [x] remove erroneous unlock
- [x] delete all historical balances in `deleteAccountRecords`

### CODE REVIEW

File: storage/modules/balance_storage.go [hunk 1]
```diff
@@ -448,7 +432,7 @@ func (b *BalanceStorage) Reconciled(
 		b.pendingReconciliationMutex.Unlock()
 	}
 
-	if err := dbTx.Set(ctx, key, new(big.Int).SetInt64(block.Index).Bytes(), true); err != nil {
+	if err := dbTx.Set(ctx, key, utils.BigIntBytes(new(big.Int).SetInt64(block.Index)), true); err != nil {
```
Comment (patrick-ogrady) [inline storage/modules/balance_storage.go:435]: 🤮 this is so ugly and inefficient

Comment (sidhujag) [reply storage/modules/balance_storage.go:435]: why not 
b := make([]byte, 8)
binary.LittleEndian.PutUint64(b, int64(block.Index))  

does bigint do something special to encode the bytes differently?

Comment (patrick-ogrady) [reply storage/modules/balance_storage.go:435]: Probably makes the most sense to just write a `GetInt64` and `StoreInt64` method for some things.

`Amount.Value` would still use `Big.Int`

Comment (sidhujag) [reply storage/modules/balance_storage.go:435]: yea maybe typedef Big.Int as Amount or something and then you can write wrappers and do optimizations as you see fit.

func (a *Amount) AsBigInt() big.Int {
	if a == nil {
		return *new(big.Int)
	}
	return big.Int(*a)
}

func (a *Amount) AsInt64() int64 {
	if a == nil {
		return 0
	}
	return (*big.Int)(a).Int64()
}

Comment (patrick-ogrady) [reply storage/modules/balance_storage.go:435]: Sorry was prototyping some stuff @sidhujag....I actually typed the response to my own message before github showed me yours.

TL;DR I found a way to fix this with a lot less work...

**Review by yflinn — APPROVED** [review]
